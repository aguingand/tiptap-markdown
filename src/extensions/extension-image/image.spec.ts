import { expect, test } from "vitest";
import { parse, serialize } from "../../../__tests__/utils";


test('parse markdown', () => {
    expect(parse('![example](example.jpg)')).toMatchSnapshot();
});
// this has no effect so should render in a paragraph
test('parse markdown inline', () => {
    expect(parse('![example](example.jpg)', { image: { inline: true } })).toMatchSnapshot();
});
test('parse html', () => {
    expect(parse('<img src="example.jpg" alt="example">')).toMatchSnapshot();
});
test.skip('serialize image', () => {
    expect(serialize('<img src="example.jpg" alt="example">')).toEqual('![example](example.jpg)');
});
