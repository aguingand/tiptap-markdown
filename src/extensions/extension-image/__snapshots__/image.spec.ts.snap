// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse html 1`] = `
[
  {
    "attrs": {
      "alt": "example",
      "src": "example.jpg",
      "title": null,
    },
    "type": "image",
  },
]
`;

exports[`parse markdown 1`] = `
[
  {
    "content": [
      {
        "attrs": {
          "alt": "example",
          "src": "example.jpg",
          "title": null,
        },
        "type": "image",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown inline 1`] = `
[
  {
    "content": [
      {
        "attrs": {
          "alt": "example",
          "src": "example.jpg",
          "title": null,
        },
        "type": "image",
      },
    ],
    "type": "paragraph",
  },
]
`;
